src/easier68k/core/opcodes/move.py

```diff
@@ -0,0 +1,216 @@
+            assert parts[1] in Move.allowed_sizes, "Size {} isn't allowed for command {}".format(parts[1], command[0])
+            size = parts[1]
+
+        # Split the parameters into EA modes
+        params = parameters.split(',')
+        assert len(params) == 2
+
+        src = EAMode.parse_ea(params[0].strip())  # Parse the source and make sure it parsed right
+        assert src.mode > EAMode.ERR, 'Error parsing src'  # -1 means error
+
+        dest = EAMode.parse_ea(params[1].strip())
+        assert dest.mode > EAMode.ERR, 'Error parsing dest'  # -1 means error
+
+        Move.__init__(self, src, dest, size)
+
+    # Include parameters here
+    def __init__(self, src: EAMode, dest: EAMode, size='Ws'):
+        # Check that the src is of the proper type (for example, can't move from an address register for a move command)
+        assert src.mode != EAMode.ARD  # Only invalid src is address register direct
+        self.src = src
+
+        # Check that the destination is of a proper type
+        assert dest.mode != EAMode.ARD and dest.mode != EAMode.IMM  # Can't take address register direct or immediates
+        self.dest = dest
+
+        # Check that this is a valid size (for example, 'MOVEA.B' is not a valid command)
+        assert size in Move.allowed_sizes
+        self.size = size
+
+    def assemble(self) -> bytearray:
+        # Create a binary string to append to, which we'll convert to hex at the end
+        tr = '00'  # Opcode
+        tr += '{0:02d}'.format(MoveSize.parse(self.size))  # Size bits
+        tr += EAModeBinary.parse_from_ea_mode_xnfirst(self.dest)  # Destination first
+        tr += EAModeBinary.parse_from_ea_mode_mfirst(self.src)  # Source second
+
+        to_return = bytearray.fromhex(hex(int(tr, 2))[2:])  # Convert to a bytearray
+        return to_return
+
+    def execute(self, simulator: M68K):
+        pass
+
+    def __str__(self):
+        # Makes this a bit easier to read in doctest output
+        return 'Move command: Size {}, src {}, dest {}'.format(self.size, self.src, self.dest)
+
+    @staticmethod
+    def is_valid(command: str, parameters: str) -> (bool, list):
+        """
+        Tests whether the given command is valid
+
+        >>> Move.is_valid('MOVE.B', 'D0, D1')[0]
+        True
+
+        >>> Move.is_valid('MOVE.W', 'D0')[0]
+        False
+
+        >>> Move.is_valid('MOVE.G', 'D0, D1')[0]
+        False
+
+        >>> Move.is_valid('MOVE.L', 'D0, A2')[0]
+        False
+
+        :param command: The command itself (e.g. 'MOVE.B', 'LEA', etc.)
+        :param parameters: The parameters after the command (such as the source and destination of a move)
+        :return: Whether the given command is valid and a list of issues/warnings encountered
+        """
+        issues = []
+        parts = command.split('.')  # Split the command by period to get the size of the command
+        assert len(parts) <= 2  # If we have more than 2 parts something is seriously wrong
```

Comment:
again here I don't actually see where you are checking to see if the command input is a MOVE or not.